portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -62,9 +67,42 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(taskEntity);
  
-    response.sendRedirect("/index.html");
+    doGet(request,response);
     }
 
+  private String getBodyData(HttpServletRequest request) throws IOException {
+    String body = null;
+    StringBuilder stringBuilder = new StringBuilder();
+    BufferedReader bufferedReader = null;
+
+    try{
+        InputStream inputStream = request.getInputStream();
+        if (inputStream != null) {
```

Comment:
Wow, this is actually quite complicated and hard to read. From a quick search, it seems like there are two methods for reading the data in the body: `getReader()` and `getInputStream()`. You aren't reading binary data, so try using `getReader()` instead, which will make the code easier to understand.

There are some examples in this stack overflow question that might work (I haven't actually tried them):
``` java
String body = request.getReader().lines()
    .reduce("", (accumulator, actual) -> accumulator + actual);
```
or
``` java
String body = request.getReader().lines().collect(Collectors.joining());
```
Link: https://stackoverflow.com/questions/14525982/getting-request-payload-from-post-request-in-java-servlet